qiskit/circuit/tools/equivalence_checker/unitary_equivalence_checker.py

```diff
@@ -0,0 +1,127 @@
+    The computation of the unitary can be done either by the quantum_info module,
+    by the Aer simulator, or by an external (non-Qiskit) simulator.
+    The comparison can either ignore or not ignore global phase.
+    """
+
+    def __init__(self, simulator, name='unitary', external_backend=None, **backend_options):
+        """
+        Args:
+            simulator (str): The type of simulator to compute the unitary.
+                Options are: 'quantum_info', 'aer', or 'external'.
+            name (str): The checker's name.
+            external_backend (BaseBackend): The backend to run,  when `simulator` is 'external'.
+            backend_options: Options to pass to the backend, when `simulator` is 'aer'
+                or 'external'.
+
+        Raises:
+            QiskitError:
+                If `simulator` is 'aer' but Aer is not installed.
+                If `simulator` ia different from 'quantum_info', 'aer', 'external'.
+        """
+
+        super().__init__(name)
+        self.simulator = simulator
+        self.backend_options = backend_options
+
+        if simulator == 'external':
+            self.backend = external_backend
+        elif simulator == 'aer':
+            try:
+                from qiskit.providers.aer import UnitarySimulator
+                self.backend = UnitarySimulator()
+                self.backend.set_options(**backend_options)
+            except ImportError:
+                raise QiskitError('Could not import the Aer simulator')
+        elif simulator == 'quantum_info':
+            self.backend = None
+        else:
+            raise QiskitError('Unrecognized simulator option: ' + str(self.simulator))
+
+    # pylint: disable=arguments-differ
+    def _run_checker(self, circ1, circ2, phase):
+        """
+        Check if circuits are equivalent.
+
+        Args:
+            circ1 (QuantumCircuit): First circuit to check.
+            circ2 (QuantumCircuit): Second circuit to check.
+            phase (str): Options are 'global' - ignoring global phase;
+                or 'equal' - not ignoring global phase.
+
+        Returns:
+            EquivalenceCheckerResult: result of the equivalence check.
+
+        Raises:
+            QiskitError:
+                If unitary creation fails (e.g., one of the circuit contains measurements,
+                    or circuits are too large).
+                If `phase` is not one of 'equal', 'global'.
+        """
+
+        # importing here to avoid circular imports
+        from qiskit.quantum_info.operators import Operator
+        from qiskit.quantum_info.operators.predicates import is_identity_matrix
+        from qiskit.compiler import transpile, assemble
+
+        equivalent = None
+        success = True
+        error_msg = None
+
+        try:
+            circ = circ1 + circ2.inverse()
+            # Optimize the circuit before creating the unitary
+            circ = transpile(circ, self.backend)
```

Comment:
For Aer method we might also want to transpile with `optimization_level=0` by default since this might be used to test transpiler passes. If you want to test transpiled circuits you can transpile them before passing to equiv checker.

Reply:
I added a `transpile_options` parameter